### PR TITLE
Add prefix arg to metric collection

### LINKED ISSUE #64: `log_dict` is missing `prefix` parameter that is mentioned in docs
## 🐛 Bug

[`MetricCollection` documentation](https://pytorch-lightning.readthedocs.io/en/1.2.0/extensions/metrics.html#metriccollection) mentions using `self.log_dict(self.train_metrics, on_step=True, on_epoch=False, prefix='train')`. The `prefix` parameter doesn't seem to be present in `log_dict` function header.

### Expected behavior

`prefix` is most likely usable in this context, so this feature should be implemented. If not - the documentation should be fixed.

### Environment

 - Lightning Version: 1.2.0

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes #64 
Adds a `prefix` argument to the metric collection for easy initialization of the same metric collection multiple times with adjusted keys
```python
mc = MetricCollection([Accuracy, Precision, Recall], prefix='train_')
out = mc(...)
# keywords of output dict will be "train_Accuracy", "train_Precision", "train_Recall"
```

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
LGTM. Can you also add return type annotations?

File: docs/source/pages/overview.rst [hunk 1]
```diff
@@ -197,25 +197,27 @@ inside your LightningModule
...
-        metrics = pl.metrics.MetricCollection(...)
-        self.train_metrics = metrics.clone()
-        self.valid_metrics = metrics.clone()
+        metrics = MetricCollection(Accuracy(), Precision(), Recall()
+        self.train_metrics = metrics.clone(prefix='train_')
+        self.valid_metrics = metrics.clone(prefix='val_')
 
     def training_step(self, batch, batch_idx):
         logits = self(x)
         ...
```
Comment (Borda) [inline docs/source/pages/overview.rst:209]:
```suggestion
        # ...
```

File: docs/source/pages/overview.rst [hunk 2]
```diff
@@ -197,25 +197,27 @@ inside your LightningModule
 
 .. code-block:: python
```
Comment (Borda) [inline docs/source/pages/overview.rst:198]:
```suggestion
.. testcode::
```

File: docs/source/pages/overview.rst [hunk 3]
```diff
@@ -197,25 +197,27 @@ inside your LightningModule
...
-        self.train_metrics(logits, y)
+        output = self.train_metrics(logits, y)
         # use log_dict instead of log
-        self.log_dict(self.train_metrics, on_step=True, on_epoch=False, prefix='train')
+        # metrics are logged with keys: train_Accuracy, train_Precision and train_Recall
+        self.log_dict(output)
 
     def validation_step(self, batch, batch_idx):
         logits = self(x)
         ...
```
Comment (Borda) [inline docs/source/pages/overview.rst:217]:
```suggestion
        # ...
```

**Review by Borda — APPROVED** [review]
